V2.0.0

JamesTheClarke opened the pull request:
**GitHub Repo Restructure**
- moved Default Modules (Coop No-Respawn) into "deprecated" since it is currently not functional
- moved out of date factions from "custom factions" to "deprecated": fixes #156 
- moved GM modules from "references" to "deprecated": fixes #148 

**Standard Faction Changes**
- added recently fixed CNTO berets to TFN GMs.
- added 3CB RPK to AR, changed AR backpack to accommodate for larger mag size, redistributed mags between AR and AAR, added 3CB RPK mags to Fireteam and Platoon Resupply Boxes: fixes #154
- changed boxguard code to ensure compatibility with A3AA update: fixes #155
- changed HMG Resupply Boxes and removed one mortar bag from mortar resupply crates to ensure they can be dragged: fixes #157
- added @Seb105 's Telestation: fixes #149
- replaced RHS flashlights with vanilla flashlights for TFN and US: fixes #147

**Default Modules Changes**
- added new CNTO TTP diary entry listing our radio channels for coop (pvp doesn't have TTP entries): fixes #153
- uniformed the naming convention of long-range channels for coop and pvp
- removed Locality Transfer (to HC) module from coop and pvp: fixes #152
- removed outdated execute code module that governed in-game chat settings (Coop only!): fixes #151

Assigned for review MMM, two members of R&D and the SSgts. Min. 2 green-light reviews needed for merger. Please only green-light if you spot-checked all three factions and the default modules. If you want to leave a review but I haven't assigned feel free to contribute nonetheless.

Once reviewed and merged I will publish the compositions via the CNTO account on Steam Workshop as well as publish the usual GitHub release.

baegel49 commented:
Just an FYI, the RPK variant used on the Russian AR cannot equip scopes. I don't know if this intended or not, but commenting anyway just to double check

JamesTheClarke commented:
> Just an FYI, the RPK variant used on the Russian AR cannot equip scopes. I don't know if this intended or not, but commenting anyway just to double check

Yeh, unless I'm mistaken this was the case previously as well with NiArms. Basically, the "old" RPKs that get drum mags don't have rail attachments. The only RKPs that can equip sights is the modern RPK-74N but that one only gets modern 60rnd stacked mags because the Russians discontinued the use of drum mags for logistical reasons.

baegel49 commented:
> > Just an FYI, the RPK variant used on the Russian AR cannot equip scopes. I don't know if this intended or not, but commenting anyway just to double check
> 
> Yeh, unless I'm mistaken this was the case previously as well with NiArms. Basically, the "old" RPKs that get drum mags don't have rail attachments. The only RKPs that can equip sights is the modern RPK-74N but that one only gets modern 60rnd stacked mags because the Russians discontinued the use of drum mags for logistical reasons.

Roger, just checking that its working as intended. I see no issues from my spot checks.

Shiny-CNTO reviewed (approved):
Looks great, this will fix quite a few issues :D

Seb105 commented:
Why not use RPK-74Ns with stick mags? It's what modern armies are doing.

JamesTheClarke commented:
> Why not use RPK-74Ns with stick mags? It's what modern armies are doing.

It's what I originally proposed in May '20 but it wasn't very popular: https://www.carpenoctem.co/forums/page/6/m/26081621/viewthread/33018467-changes-to-standard-factions-pure-balance-vs-true-life/post/138798845#p138798845

Considering this is a fairly urgent update and previous push back against 60rnd stacked mags I'd say such a switch should be discussed on the forums (Staff or MMT) before changing to 60rnd stacked mags.

freghar commented:
I know this is a bit late, but are you sure you haven't accidentally missed any "Execute Code (Eden)" modules by forgetting to place the "inhibit" module before placing/editing/saving the module compositions?

I'm pretty sure there used to be an "Execute Code (Eden)" module that would set up respawn and other mission attributes. That doesn't seem to be the case anymore.

freghar commented:
Also, the names of compositions aren't showing up in the list in editor, they appear greyed out and nameless.
This is possibly caused by the compositions being saved on a non-main modset (ie. with campaign repo, I see `Desert` mentioned).

The fix is to clear out `requiredAddons[]` in `header.sqe` of each composition (and ideally do so on any composition update), ie.
```
version=54;
name="Default Modules (Coop)";
author="CNTO Mission Making Team";
category="CNTODefaults";
requiredAddons[]={};
```
instead of
```
version=54;
name="Default Modules (Coop)";
author="CNTO Mission Making Team";
category="CNTODefaults";
requiredAddons[]=
{
	"A3_Characters_F",
	"A3_Data_F_Exp_A_Virtual",
	"acre_api",
	"Desert",
	"A3_Modules_F_Intel",
	"a3aa_ee_modules",
	"a3aa_ee_basic_vehicle_cargo",
	"a3aa_ee_arsenal_respawn"
};
```

JamesTheClarke commented:
> I know this is a bit late, but are you sure you haven't accidentally missed any "Execute Code (Eden)" modules by forgetting to place the "inhibit" module before placing/editing/saving the module compositions?
> 
> I'm pretty sure there used to be an "Execute Code (Eden)" module that would set up respawn and other mission attributes. That doesn't seem to be the case anymore.

Damn, in that case, we don't have that in our composition anymore since the switch to A3AA mid-last year. Which explains why the PvP Composition is no longer functional.  I remembered it being around too but didn't realize it was one of those modules that needed an inhibitor.

Could you post the code here? I'll open a new issue for the next update to fix this.

JamesTheClarke commented:
> Also, the names of compositions aren't showing up in the list in editor, they appear greyed out and nameless.

This issue was fixed on my end by completely deleting the composition folders and then pasting the new ones in. Before I had the same issue of compositions being greyed out. I didn't run anything besides the main repo when creating the update.

freghar commented:
> Damn, in that case, we don't have that in our composition anymore since the switch to A3AA mid-last year.

I'm not aware of anything that changed mid-last year relevant to this, it [has been like this](https://github.com/freghar/arma-additions/commit/833d2a5f6) ever since we introduced module compositions instead of the mod-based cnto-units (we had a transitional period when modules were [being added by a menu item](https://github.com/CntoDev/cnto-units/blob/master/addons/shared/functions/mission_defaults/fn_placeDefaultModules.sqf)).

Maybe we should add some chat message into the module code, so that it's clear there's some logic running,
```
systemChat "Restoring/overwriting mission settings for Coop.";
```
> Could you post the code here? I'll open a new issue for the next update to fix this.

I'm not on Windows to check, but just place an older version of the compositions, that should still work even if they're greyed out (you can tell them apart by the tooltip).


> > Also, the names of compositions aren't showing up in the list in editor, they appear greyed out and nameless.
> 
> This issue was fixed on my end by completely deleting the composition folders and then pasting the new ones in. Before I had the same issue of compositions being greyed out. I didn't run anything besides the main repo when creating the update.

Yeah, you can just empty `requiredAddons[]={}` in `header.sqe` next time, might be easier.

freghar commented:
Maybe we should also mention the inhibitor thing in [the README file, guide on how to update compositions](https://github.com/CntoDev/compositions#how-to-maintain), but I've noticed people don't seem to follow it anymore, because there's been a lot of noise in the changes and a lot of useless properties and checkboxes with default values (which don't hurt, but increase mission size).

JamesTheClarke commented:
> I'm not aware of anything that changed mid-last year relevant to this, it [has been like this](https://github.com/freghar/arma-additions/commit/833d2a5f6) ever since we introduced module compositions instead of the mod-based cnto-units (we had a transitional period when modules were [being added by a menu item](https://github.com/CntoDev/cnto-units/blob/master/addons/shared/functions/mission_defaults/fn_placeDefaultModules.sqf)).

I'm referring to when we updated A3AA in our repo mid-last year, an update that introduced the need for the inhibitor module. Before that the code modules would not disappear automatically. The issue with the inhibitor module requirement is that unless you know which compositions are affected it's very easy to forget to place the inhibitor module first when updating the composition. This happened this year with Highway when he updated the camo compositions and probably with me or Seb last year when we updated all our compositions to be compatible when we switched to the A3AA update that introduced the inhibitor module.

JamesTheClarke commented:
> Maybe we should also mention the inhibitor thing in [the README file, guide on how to update compositions](https://github.com/CntoDev/compositions#how-to-maintain), but I've noticed people don't seem to follow it anymore, because there's been a lot of noise in the changes and a lot of useless properties and checkboxes with default values (which don't hurt, but increase mission size).

I was thinking the same today and actually would prefer if we shorten the readme file and move documentation to the wiki which doesn't need commits to be updated.

Seb105 commented:
> Damn, in that case, we don't have that in our composition anymore since the switch to A3AA mid-last year

Not true, it got broken in the most recent update. It's now causing problems such as AI not being disabled in multiplayer lobby, which worked fine for a while.

Seb105 commented:
https://github.com/CntoDev/compositions/pull/166

JamesTheClarke commented:
> > Damn, in that case, we don't have that in our composition anymore since the switch to A3AA mid-last year
> 
> Not true, it got broken in the most recent update. It's now causing problems such as AI not being disabled in multiplayer lobby, which worked fine for a while.

Hmm, I assumed we updated the default modules back then because the one-life respawn module has been missing / not working for a long time, but I guess this must be a different issue then. Thanks for the hotfix PR #166 and the issue report #165 

Gonna review it now and deploy it tonight unless there's any objections.
